### ISSUE TITLE
Add ability to set default backupstoragelocation

### ISSUE THREAD

**Issue by sharkymcdongles** [vmware-tanzu/helm-charts#251]
**Describe the problem/challenge you have**
1.6.0 introduced a default setting to the crd:

https://github.com/vmware-tanzu/helm-charts/blob/main/charts/velero/crds/backupstoragelocations.yaml#L26

They say to set this flag via velero cli, but this is not good practice if chart can do it for us by exposing this under the configuration.backupStorageLocation block.

**Describe the solution you'd like**
Add ability to set this to true or false.

**Comment by sharkymcdongles** [comment]
I found a workaround for now to those that are interested via an initContainer, but I am sure we can do this in a cleaner way:

```
initContainers:
- image: velero/velero:v1.6.0
  imagePullPolicy: IfNotPresent
  name: velero-set-bsl
  command: ['sh', '-c', '/velero backup-location set gcp --default']

```

**Comment by jenting** [comment]
You should be able to configure the BSL as the **default** like this

```yaml
configuration:
  provider: gcp
  backupStorageLocation:
    name: gcp
    provider: gcp
    bucket:
    prefix:
    default: true
```

**Comment by sharkymcdongles** [comment]
The way the templating is set, if the default isn't explicitly referred to from the values like you suggested it won't be templated into the final resource:
https://github.com/vmware-tanzu/helm-charts/blob/main/charts/velero/templates/backupstoragelocation.yaml

Ideally this chart should move to a non hardcoding setup where it takes all keys under the backupStorageLocation block then places it there enmasse without assigning it to specific values to prevent the need for future adjustments.

**Comment by sharkymcdongles** [comment]
FYI: for my fix to work this needs to be merged or CRDs upgraded manually as the chart fails to do so for you:

https://github.com/vmware-tanzu/helm-charts/pull/246

**Comment by jenting** [comment]
> Ideally this chart should move to a non hardcoding setup where it takes all keys under the backupStorageLocation block then places it there enmasse without assigning it to specific values to prevent the need for future adjustments.

I had a PR for this https://github.com/vmware-tanzu/helm-charts/pull/243 😉

**Comment by jenting** [comment]
The PR merged and release along with 2.18.2.
If no further concern, I think this issue could be closed?
